src/classes/BGE_Wizard_CTRL.cls

```diff
@@ -0,0 +1,307 @@
+    * @return list of Template Fields with sObjectName and Name.
+    */
+    public static List<AvailableField> getAvailableFields() {
+
+        List<AvailableField> fields = new List<AvailableField>();
+
+        Map<String, Schema.DescribeFieldResult> fieldMap = UTIL_Describe.getAllFieldsDescribe(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'));
+        Set<String> bannedFields = new Set<String>{'Account1Imported__c',
+                'Account1ImportStatus__c',
+                'Account2Imported__c',
+                'Account2ImportStatus__c',
+                'ApexJobId__c',
+                'Contact1Imported__c',
+                'Contact1ImportStatus__c',
+                'Contact2Imported__c',
+                'Contact2ImportStatus__c',
+                'DonationImported__c',
+                'DonationImportStatus__c',
+                'Donation_Possible_Matches__c',
+                'FailureInformation__c',
+                'HomeAddressImported__c',
+                'HomeAddressImportStatus__c',
+                'HouseholdAccountImported__c',
+                'ImportedDate__c',
+                'NPSP_Data_Import_Batch__c',
+                'PaymentImported__c',
+                'PaymentImportStatus__c',
+                'Payment_Possible_Matches__c',
+                'Status__c',
+                'Account2ImportStatus__c'
+        };
+
+        Set<String> allowedObjects = new Set<String>{
+                'Opportunity',
+                'Payment'
+        };
+
+        // these fields are required only if at least 1 field from its object group is selected
+        Set<String> systemRequiredFields = new Set<String>{
+                'Account1_Name__c',
+                'Account2_Name__c',
+                'Donation_Amount__c',
+                'Contact1_LastName__c',
+                'Contact2_LastName__c'
+        };
+
+        // looping through the fields on DataImport__c
+        for (Schema.DescribeFieldResult dfr : fieldMap.values()) {
+            String dataImportFieldApiName = UTIL_Namespace.StrTokenRemoveNSPrefix(dfr.name);
+            //Special case for Donation_Donor because it's derived
+
+            //todo: rework these 2 lines
+            String dataImportFieldMappedObject = dfr.inlineHelpText != null ? dfr.inlineHelpText.split('\\.')[0] : null;
+            Boolean fieldAllowed = allowedObjects.contains(dataImportFieldMappedObject) && !bannedFields.contains(dataImportFieldApiName) && dfr.inlineHelpText != null;
+
+            if (fieldAllowed) {
+
+                AvailableField fld = new AvailableField();
+
+                // TODO: make these localized with schema describes
+                fld.sObjectName = dataImportFieldMappedObject;
+
+                // TODO: pull these labels from target fields' schema based on help text mapping
+                fld.label = (dfr.getLabel().contains(' ') && dfr.getLabel().split(' ',2)[0] == fld.sObjectName) ?
+                        dfr.getLabel().split(' ',2)[1] :
```

Comment:
Why the double indentation?

Reply:
I think Beth put that in, but I think to see the if/else matching lines separately. I refactored the ternary to be more readable with a boolean.

Reply:
I don't mind using a ternary, but I think the intent of the logic could use explanation. What is this trying to do?

```
(dfr.getLabel().contains(' ') && dfr.getLabel().split(' ',2)[0] == fld.sObjectName) ?
                        dfr.getLabel().split(' ',2)[1] : 
```